### PR TITLE
auth files

### PR DESCRIPTION
**Pull request by jeremywbrazell**
Created auth files for basic and bearer authentication.  Pushing up for review.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,8 +22,21 @@
...
+    "express": "^4.17.1",
+    "jsonwebtoken": "^8.5.1",
+    "method-override": "^3.0.0",
+    "morgan": "^1.10.0",
+    "nodemon": "^2.0.15",
+    "pg": "^8.7.1",
+    "sequelize": "^6.12.0-beta.1",
+    "sequelize-cli": "^6.3.0",
+    "sqlite3": "^5.0.2",
+    "superagent": "^6.1.0"
```
Comment (MuckT) [inline package.json]: Think a few of these can move to dev dependencies like `nodemon`. I'd also suggest we remove `morgan` and use a different logger if possible. Let's also see if we can get `axios` working without method override.

File: src/error-handlers/acl.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict'
```
Comment (MuckT) [inline src/error-handlers/acl.js]: This file looks like a duplicate of the one in middleware. Lets delete it.

File: src/middleware/basic.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  let [user, pass] = base64.decode(basic).split(':')
+
+  try {
+    req.user = await users.authenticateBasic(user, pass)
+    next()
+  } catch (e) {
+    _authError()
+  }
+
+  function _authError() {
```
Comment (MuckT) [inline src/middleware/basic.js]: lets rename this function to remove the `_` as this doesn't appear to be recursive. If it's use commonly we can keep it, but haven't seen this before.

File: src/middleware/bearer.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    const token = req.headers.authorization.split(' ').pop()
+    const validUser = await users.authenticateToken(token)
+    req.user = validUser
+    req.token = validUser.token
+    next()
+  } catch (e) {
+    _authError()
+  }
+
+  function _authError() {
```
Comment (MuckT) [inline src/middleware/bearer.js]: Same comment as with basic around `_authError()`.

File: src/models/modelsIndex.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+'use strict'
+
+const { Sequelize, DataTypes } = require('sequelize')
+const investorModel = require('./investor.js')
```
Comment (MuckT) [inline src/models/modelsIndex.js]: Please remove investor model.

**Review by MuckT — CHANGES_REQUESTED** [review]

**Review by MuckT — APPROVED** [review]
LGMT!

**Review by fadnesscharlie — APPROVED** [review]
I approve

**Pull request merged**